### ISSUE TITLE
PrimeFaces cannot find default message bundles when deployed as an OSGi bundle

### ISSUE THREAD

**Issue by stiemannkj1** [primefaces/primefaces#3878]
Getting messages from the default `ResourceBundle`s fails in an OSGi environment because the default `ResourceBundle`s cannot be found. For example, the following calls will return `null` when called from an OSGi WAB or bundle that depends on the PrimeFaces bundle:

```
MessageFactory.getMessage(UIData.ARIA_NEXT_PAGE_LABEL, new Object[]{});
MessageFactory.getMessage("javax.faces.converter.DateTimeConverter.DATE", FacesMessage.SEVERITY_ERROR, params);
```

The reason these calls return `null` because `MessageFactory` uses the `Thread.currentThread().getContextClassLoader()` to look for the bundles. In OSGi, that `ClassLoader` is the loader of the calling WAB, but it does not have access to the default `ResourceBundle`s that are contained in PrimeFaces, Mojarra, and MyFaces.

In order to fix this problem, `MessageFactory` should look for `ResourceBundle`s using both the thread-context loader and the loader of the bundle containing the default resource (such as the PrimeFaces bundles' loader or Mojarra's loader).

**Comment by stiemannkj1** [comment]
I can't self-assign this issue, but I'm planning on sending a fix for this today.

**Comment by melloware** [comment]
@stiemannkj1 That is what I do I usually say "PR to follow" at the bottom of my issue.  Nice work pioneering all this OSGI stuff!

**Comment by stiemannkj1** [comment]
Thanks @melloware :) I've actually got PrimeFaces 6.2 working with OSGi, but it requires bytecode weaving (basically an elegant, relatively safe hack). If #3880 is merged, PrimeFaces 6.3 should work in OSGi without any hacks.
